src/popup/utils/getPopupProps.ts

```diff
@@ -0,0 +1,90 @@
+import { v4 as genUuid } from 'uuid';
+import { popupProps } from './testsConfig';
+import { buildTx, CONNECTION_TYPES } from './index';
+import { IS_EXTENSION, POPUP_TYPE } from '../../lib/environment';
+import '../../lib/initPolyfills';
+
+interface PopupMessageData {
+  type: 'resolve' | 'reject' | 'getProps'
+  payload?: any
+}
+
+type PopupMessageProps = Record<string, any>;
+
+const postMessage = (() => {
+  const pendingRequests: Record<string, any> = {};
+  let background: any;
+
+  return async ({ type, payload }: PopupMessageData): Promise<PopupMessageProps> => {
+    if (!IS_EXTENSION || !browser) {
+      throw new Error('Supported only in browser extension');
+    }
+
+    if (!background) {
+      background = await browser.runtime.connect({ name: CONNECTION_TYPES.POPUP });
+      background.onMessage.addListener(({ uuid, res }: any) => {
+        if (!pendingRequests[uuid]) {
+          throw new Error(`Can't find request with id: ${uuid}`);
+        }
+        pendingRequests[uuid].resolve(res);
```

Comment:
```suggestion
        pendingRequests[uuid].resolve(res);
        delete pendingRequests[uuid];
```

I guess we need to delete resolved requests from memory

Reply:
Sure, makes sense.